include/nanobind/stl/detail/nb_set.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <nanobind/nanobind.h>
+
+NAMESPACE_BEGIN(NB_NAMESPACE)
+NAMESPACE_BEGIN(detail)
+
+template <typename Value_, typename Key> struct set_caster {
+    NB_TYPE_CASTER(Value_, const_name("set[") + make_caster<Key>::Name + const_name("]"));
+
+    using KeyCaster = make_caster<Key>;
+
+    bool from_python(handle src, uint8_t flags, cleanup_list *cleanup) noexcept {
+        value.clear();
+
+        PyObject* iter = obj_iter(src.ptr());
+        if (iter == nullptr) {
+            PyErr_Print();
```

Comment:
`PyErr_Clear()`, not `PyErr_Print()`. The type casters should not print messages in the case of failure.

Reply:
Replaced. Thanks.